delta/service/src/main/scala/ch/epfl/bluebrain/nexus/delta/service/projects/ProjectsImpl.scala

```diff
@@ -160,20 +160,35 @@ object ProjectsImpl {
     KeyValueStore.distributed(moduleType, clock)
   }
 
+  @SuppressWarnings(Array("IsInstanceOf"))
   private def startIndexing(
       config: ProjectsConfig,
       eventLog: EventLog[Envelope[ProjectEvent]],
       index: ProjectsCache,
-      projects: Projects
+      projects: Projects,
+      si: Set[ScopeInitialization]
   )(implicit as: ActorSystem[Nothing], sc: Scheduler) =
     StreamSupervisor(
       "ProjectsIndex",
       streamTask = Task.delay(
         eventLog
           .eventsByTag(moduleType, Offset.noOffset)
-          .mapAsync(config.indexing.concurrency)(envelope =>
-            projects.fetch(envelope.event.project).redeemCauseWith(_ => IO.unit, res => index.put(res.value.ref, res))
-          )
+          .mapAsync(config.indexing.concurrency) { envelope =>
+            projects
+              .fetch(envelope.event.project)
+              .redeemCauseWith(
+                _ => IO.unit,
+                { resource =>
+                  index.put(resource.value.ref, resource) >>
+                    IO.when(!resource.deprecated && envelope.event.isInstanceOf[ProjectCreated]) {
```

Comment:
Why the two checks? it is enough with the second. Although it is pretty ugly. I would rather introduce a method on `ProjectEvent`

```scala
def isCreated: Boolean  =
self match {
   case _: ProjectCreated => true
   case _ => false
```